python/ray/reporter.py

```diff
@@ -222,10 +227,10 @@ def perform_iteration(self):
     def run(self):
         """Publish the port."""
         thread_pool = futures.ThreadPoolExecutor(max_workers=10)
-        server = grpc.server(thread_pool, options=(("grpc.so_reuseport", 0), ))
+        server = grpc.server(thread_pool)
         reporter_pb2_grpc.add_ReporterServiceServicer_to_server(
             ReporterServer(), server)
-        port = server.add_insecure_port("[::]:0")
+        port = server.add_insecure_port("[::]:{}".format(self.port))
```

Comment:
What happened if the port conflicts?

Reply:
Port will be selected inside `ray.init`. There's still possible race condition (which is that we don't find a port when we bind them), but it will be just same as other processes. We'd need to find a good way to avoid this, but that's the out of scope of this PR.